### PR TITLE
Deployment on K8s

### PR DESCRIPTION
**Pull request by kkowalski**
1. Introducing build-test-deploy workflow, I've removed ci and release workflows.
2. Deployment to staging sets up a new local ethereum node to which it connects so each deploys wipes the state
4. Deployment to prod currently points at Goerli testnet
5. Deployments require manual approval in the workflow

### CODE REVIEW

**Review by philsippl — DISMISSED** [review]

File: deploy/geth/values.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  # requests:
+  #   cpu: 100m
+  #   memory: 128Mi
+
+autoscaling:
+  enabled: false
+  minReplicas: 1
+  maxReplicas: 100
+  targetCPUUtilizationPercentage: 80
+  # targetMemoryUtilizationPercentage: 80
```
Comment (mircobordoni) [inline deploy/geth/values.yaml:79]: should this be enabled?

File: .github/workflows/build-test-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+        run: helm upgrade "${{ github.event.repository.name }}" ./deploy/hubble
+          --install --atomic --timeout 60s
+          --namespace "${{ github.event.repository.name }}"
+          --set image.tag="${{ github.sha }}"
+          --set ingress.host=hubble.stage-crypto.worldcoin.dev
+          --set environment=stage
+  deploy-prod:
+    runs-on: ubuntu-latest
+    needs: [ build-and-push, e2e-tests ]
+#    if: github.ref == 'refs/heads/master'
```
[the remark this answers is not in the document]
Comment (kkowalski) [reply .github/workflows/build-test-deploy.yml:320]: yes, it should

File: deploy/geth/values.yaml (hunk 1 above)
Comment (kkowalski) [reply deploy/geth/values.yaml:79]: Geth chart is only used for staging and this came commented out from `helm create chart`. We will never use autoscaling for this so I'll just leave it as is.

File: .github/workflows/build-test-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+      - name: Submit to codecov.io
+        uses: codecov/codecov-action@v2.1.0
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }} # Optional for public repos
+          files: ./coverage.out
+          flags: ${{ github.job }}
+          fail_ci_if_error: true
+          verbose: true
+  build-and-push:
+    needs: [test]
```
[the remark this answers is not in the document]
Comment (kkowalski) [reply .github/workflows/build-test-deploy.yml:173]: e2e-tests are a dependency on the prod deploy but I thought lets skip it for staging deploy

**Review by mircobordoni — APPROVED** [review]

**Pull request merged**